File: JAVA/Day4/GroceryManagerApp.java
import java.util.ArrayList;
import java.util.Scanner;

public class GroceryManagerApp {

    public static ArrayList<String> groceryList = new ArrayList<>();
    public static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        boolean running = true;

        System.out.println("What is your name? ");
        String userName = scanner.nextLine().toUpperCase();
        System.out.println("Welcome Mr " + userName + " to your personal grocery app");
        System.out.println("\n|=-=-=-= " + userName + " Manager App =-=-=-=|");

        while (running) {
            showMenu();
            int choice = getUserChoice();

            switch (choice) {
                case 1 -> addItem();
                case 2 -> removeItem();
                case 3 -> showItems();
                case 0 -> {
                    running = false;
                    System.out.println("Exiting Grocery Manager App. Goodbye!");
                }
                default -> System.out.println("Invalid choice. Please select a valid option.");
            }
        }
    }

    public static void showMenu() {
        System.out.println("\nWhat would you like to do today on your grocery app?");
        System.out.println("1. Add an Item");
        System.out.println("2. Remove an Item");
        System.out.println("3. Show All Items");
        System.out.println("0. Exit the grocery app");
    }

    public static int getUserChoice() {
        try {
            return Integer.parseInt(scanner.nextLine());
        } catch (NumberFormatException e) {
            System.out.println("Dear Sir, this is an invalid input. Please enter a number.");
            return -1;
        }
    }

    public static void addItem() {
        System.out.print("Enter any item of your choice to add: ");
        String itemToAdd = scanner.nextLine();
        groceryList.add(itemToAdd);
        System.out.println("'" + itemToAdd + "' added successfully!");
    }

    public static void removeItem() {
        System.out.print("Enter item to remove: ");
        String itemToRemove = scanner.nextLine();
        if (groceryList.remove(itemToRemove)) {
            System.out.println("'" + itemToRemove + "' removed successfully!");
        } else {
            System.out.println("'" + itemToRemove + "' not found in the list.");
        }
    }

    public static void showItems() {
        System.out.println("\n--- Grocery List ---");
        if (groceryList.isEmpty()) {
            System.out.println("No items in the list.");
        } else {
            for (int i = 0; i < groceryList.size(); i++) {
                System.out.println((i + 1) + ". " + groceryList.get(i));
            }
        }
    }
}

File: Python/Day4/GroceryManagerApp.py
grocery_list = []



def show_menu():
    print("\nWhat would you like to do today on your grocery app?")
    print("1. Add an Item")
    print("2. Remove an Item")
    print("3. Show All Items")
    print("0. Exit the grocery app")



def get_user_choice():
    try:
        return int(input("Enter choice: "))
    except ValueError:
        print("Dear Sir, this is an invalid input. Please enter a number.")
        return -1



def add_item():
    item = input("Enter any item of your choice to add: ")
    grocery_list.append(item)
    print(f"'{item}' added successfully!")



def remove_item():
    item = input("Enter item to remove: ")
    if item in grocery_list:
        grocery_list.remove(item)
        print(f"'{item}' removed successfully!")
    else:
        print(f"'{item}' not found in the list.")



def show_items():
    print("\n=-=-=-= Grocery List =-=-=-=")
    if not grocery_list:
        print("No items in the list.")
    else:
        for i, item in enumerate(grocery_list, start=1):
            print(f"{i}. {item}")








def main():
    print("What is your name?")
    user_name = input().upper()
    print(f"Welcome Mr {user_name} to your personal grocery app")
    print(f"\n|=-=-=-= {user_name} Manager App =-=-=-=|")

    running = True
    while running:
        show_menu()
        choice = get_user_choice()

        if choice == 1:
            add_item()
        elif choice == 2:
            remove_item()
        elif choice == 3:
            show_items()
        elif choice == 0:
            print("Exiting Grocery Manager App. Goodbye!")
            running = False
        else:
            print("Invalid choice. Please select a valid option.")
